sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/ManagedHsmImpl.java

```diff
@@ -0,0 +1,219 @@
+import com.azure.core.http.HttpPipeline;
+import com.azure.core.http.rest.PagedFlux;
+import com.azure.core.http.rest.PagedIterable;
+import com.azure.core.http.rest.Response;
+import com.azure.core.http.rest.SimpleResponse;
+import com.azure.resourcemanager.keyvault.KeyVaultManager;
+import com.azure.resourcemanager.keyvault.fluent.models.ManagedHsmInner;
+import com.azure.resourcemanager.keyvault.fluent.models.PrivateEndpointConnectionInner;
+import com.azure.resourcemanager.keyvault.models.Keys;
+import com.azure.resourcemanager.keyvault.models.ManagedHsm;
+import com.azure.resourcemanager.keyvault.models.ManagedHsmSku;
+import com.azure.resourcemanager.keyvault.models.MhsmNetworkRuleSet;
+import com.azure.resourcemanager.keyvault.models.PrivateEndpointServiceConnectionStatus;
+import com.azure.resourcemanager.keyvault.models.PrivateLinkServiceConnectionState;
+import com.azure.resourcemanager.keyvault.models.PublicNetworkAccess;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.PrivateLinkResource;
+import com.azure.resourcemanager.resources.fluentcore.arm.models.implementation.GroupableResourceImpl;
+import com.azure.resourcemanager.resources.fluentcore.utils.PagedConverter;
+import com.azure.resourcemanager.resources.fluentcore.utils.ResourceManagerUtils;
+import com.azure.security.keyvault.keys.KeyAsyncClient;
+import com.azure.security.keyvault.keys.KeyClientBuilder;
+import com.azure.security.keyvault.keys.KeyServiceVersion;
+import reactor.core.publisher.Mono;
+
+import java.time.OffsetDateTime;
+import java.util.Collections;
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Implementation for ManagedHsm and its parent interfaces.
+ */
+class ManagedHsmImpl
+    extends GroupableResourceImpl<ManagedHsm, ManagedHsmInner, ManagedHsmImpl, KeyVaultManager>
+    implements ManagedHsm {
+    private KeyAsyncClient keyClient;
+    private final HttpPipeline mhsmHttpPipeline;
+
+    ManagedHsmImpl(String name, ManagedHsmInner innerObject, KeyVaultManager manager) {
+        super(name, innerObject, manager);
+        mhsmHttpPipeline = manager().httpPipeline();
+        if (innerModel().properties() != null && innerModel().properties().hsmUri() != null) {
+            keyClient = new KeyClientBuilder()
+                .pipeline(mhsmHttpPipeline)
+                .vaultUrl(innerModel().properties().hsmUri())
+                .serviceVersion(KeyServiceVersion.V7_2)
+                .buildAsyncClient();
+        }
+    }
+
+    /**
+     * (Internal use only)
+     * We haven't supported creating {@link ManagedHsm} in convenience layer yet. This is for implementing necessary abstract class contract.
+     * @return {@link Mono} of the created {@link ManagedHsm} instance
+     */
+    @Override
+    public Mono<ManagedHsm> createResourceAsync() {
+        return manager().serviceClient().getManagedHsms().createOrUpdateAsync(resourceGroupName(), name(), innerModel())
+            .map(innerToFluentMap(this));
```

Comment:
nit, you may add a comment here about init after create (it helps when one add create feature).
https://github.com/Azure/azure-sdk-for-java/blob/9ed0d0b981e78816978866098f3cc7d10a3e15a8/sdk/resourcemanager/azure-resourcemanager-keyvault/src/main/java/com/azure/resourcemanager/keyvault/implementation/VaultImpl.java#L398

Reply:
wait a minute, do we create this?

Reply:
No, it's not exposed through our interface `ManagedHsm`. It's inherited from implementation abstract class `GroupableResourceImpl`. 

I added `(Internal use only)` to the method and user should have no access as it's in implementation and package and package private.

Reply:
I mean, does this method ever called in runtime?

Reply:
No, currently this method is never used at runtime.

Reply:
If it is never used, I kind of prefer to comment it out, and still use e.g.
```java
throw new UnsupportedOperationException("method [createResourceAsync] not implemented in class [com.azure.resourcemanager.keyvault.implementation.ManagedHsmImpl]");
```

Reply:
Got it. Changed.